### DISCUSSION TITLE
Hardware Catalog - Please post your working (or not working) hardware setups here

### DISCUSSION THREAD

**Discussion by WheezyE** [WheezyE/Winelink#25] [General]
I just wanted to catalog which hardware setups we all have working with RMS Express + wine + wine-mono + box86.

I'll go first. KI7POL:
FT-891+dummyload, RPi4(4GB): VARA HF, VARA FM, VARA Chat, and ARDOP with PTT & Channel Selection CAT control.

 - Have not tested OTA connections.
 - FT-891 is set to 9600 baud onboard. USB is connects RPi4 and radio. Using COM1 port.
 - VARA FM & VARA Chat requires me to select "FT-991" with RTS+DTR enabled.

**Comment by WheezyE** [comment]
F4GMU:
RaspberryPi400, IC-7100, modem pactor Swiss PTC-II: VARA HF and Pactor
https://github.com/WheezyE/Winelink/issues/17#issuecomment-1028622165

**Reply by akopac** [reply]
Hi F4GMU
I have your same setup for my Nav system on my cruising boat. I am currently anchored in Nuku Hiva FP. Anyhow. We’re you able to get this to work with your pi400 and the Pactor II MODEM. My modem is a Pactor II PRO flashed to Pactor III and does rig control on my icom 718.   Reall really appreciate your feedback. Does propagation, rig control, saildocs catalog, and gribs work?  Thanks so much for you feedback. KG6AMH,73s

**Comment by WheezyE** [comment]
KD2ROS:
RPi 4 8GB, IC7100 (native IC7100 sound card for both VARA FM and VARA HF), bluetooth Mobilink’d TNC3 for Packet Winlink: VARA FM, VARA HF and Packet Winlink
https://github.com/WheezyE/Winelink/issues/17#issuecomment-1030624391

**Comment by WheezyE** [comment]
IC-7100, Vara FM/HF, packet with direwolf (build-a-pi)
PD7RON

**Comment by mattybaus** [comment]
9V1MH/VK2AMH/YF2AAH

Trying a few things - IC-7610, X-6100, IC-705, TX-500, KX-2, KX-3 - with Mobilink, Signalink (USB), DigiRig and of course the inbuilt interfaces in the 705, 7610 and x6100 - will let you know how it goes

Be patient but this is a great project and if we can get all the other software to run as well (FLDigii, WSJT-X, JS8Call etc) then it does exactly what a lot of us want - not having a *nix Vara option means we have all gone to a tablet/miniPC and abandoned the RPi.

Would also like to know if you are looking at the Pi 64bit OS for use on a Pi4/Zero2W?
73 Matt

**Reply by WheezyE** [reply]
Hey Matt, sorry I didn’t see the question at the end of your post. I’m interested in how this might run on a Pi0 2W, but I’m not confident that it will run well enough to be usable. I’ll let you know if I get a Pi0 2W to play with. Best of luck with your setup too

**Comment by Wes-KF7BWS** [comment]
I am using Raspberry PI 4 8 gig 128 gig Micro SD, 64 bit buster, over clocked to 1500, TIMEWAVE Navigator attached to a USB 3 hub connected to the Pi in the USB 3 slot. The Navigator is connected to an Elecraft K3. Under Buster all native Linux software works very good.  been using WSJT-X for a couple of weeks no problems. 

Installed Winlink Express and VARA HF, FM, Sat. software runs OK. CAN NOT GET THE COM PORTS IN VARA HF WINLINK SESSION TO MAP TO THE TTYUSBX PORTS. I have made links using ln, edited the .reg file edited the registry regedit. nothing works.

IF i don't have RIG control and no audio out, so what if the software runs.
Wes Larson 
KF7BWS

**Reply by SpudGunMan** [reply]
same issue IC705 rpi4 and everything else works but winlink-VARA-HF  same as this past comment (CAN NOT GET THE COM PORTS IN VARA HF WINLINK SESSION TO MAP TO THE TTYUSBX PORTS. I have made links using ln, edited the .reg file edited the registry regedit. nothing works.) 


I want to be clear this very specific to -winlink and Vara-HF-PTT - it is the only non working (I assume a winlink issue with wine) FM and VarAC and VARChat HF all work fine. (similar but I dont think the same .. could be a ocx issue https://github.com/WheezyE/Winelink/issues/17)

**Reply by Wes-KF7BWS** [reply]
I have the very same issue.

My set up is an Elecraft K3 a Timewave Navigator and a Raspberry Pi 4 8 gig with dual touch screens. Everything that is native Linux which I have set up works fine. I have been using WSJT-X for several weeks with very little too no trouble. 

Everything that runs through wine does not work. I can not get a link from the windows com or USB ports to the actual device in Linux. I have tried the same things you have, regedit, ln links, patching the reg file etc.

For me the Winlink and VARA apps are the most important. This issue however will effect any windows based com application.

WSJT-X was written so that it could be compiled for both Windows and Linux, the same needs to be done for Winlink and VARA. Nothing beats running native. II have the same problems with my logging software from N3FJP.

**Reply by WheezyE** [reply]
Hello, my email alert hasn't been updating me on the discussions page, just the issues trackers - sorry for the delay.

> IF i don't have RIG control and no audio out, so what if the software runs

I'm also sorry for your disappointment with the script. This gihub page is made for free/opensource collaboration though, so if you find a solution before I do, let me know. Best of luck

**Reply by WheezyE** [reply]
>  could be a ocx issue #17

Good thinking, but that bug was fixed by wine-mono's author and all mono's going forward have that function implemented.
Off the top of my head, it could maybe be a permissions issue with the USB dialout group? I'll have to properly test this when I get some more time.

**Reply by Wes-KF7BWS** [reply]
I may not have made my self clear. I DO get all audio out of the PI to the modem a TIMEWAVE Navigator ergo audio to the RIG. I DO NOT however get Rig control through to the Navigator. from the PI. 

My Pi has since died and I have gone back to my windows machine. Same problem. Before implementing the Pi my windows machine worked flawlessly. I need to find out why my Pi died.

**Comment by PD7RON** [comment]
Hello All
Earlier I have posted that on my ic7100 all the modes were working with
patmenu and rms-express.
That was with the version where the icons were on the desktop.
The version with the icons in the menu is not working so well for me.
For the vara-hf it works for me with patmenu. But when I try to work
vara-fm with patmenu it is starting and connecting but not always there is
an audio signal. And because of that, I can`t make a properly connection.
Listening to another radio, I am getting a carrier, but there is no audio.
The option to work with vara fm is to use rms-express that works well for
me. But the vara HF can't find the com port under rms-express and so I get
an error with no connection to the ic 7100.
Conclusion: vara hf works for me with patmenu and vara-fm works for me with
the use of rms-express.
That is not handy, and I hope that someone knows a solution that it will
work again with both or with patmenu or rms-express so changing isn't
necessary anymore.
I am using debian bullseye 32bit and have also tried 64 bit on a raspberry
pi 4. Both debian`s gave the same problems.

73 Ron
PD7RON

Op do 18 aug. 2022 om 20:49 schreef Wes-KF7BWS ***@***.***>:

**Reply by WheezyE** [reply]
This might be from changing to a different version of box86 in the newer version of the script.  It is always possible to go to an older version of the script using the Clock icon on the main page, then going to the script, then going to 'file', then going to 'RAW', and getting a link to download (here's a link to a "raw" version of I think the script that put the icons on the desktop: https://raw.githubusercontent.com/WheezyE/Winelink/5d4d6f83023eb9e7b02ec783a9b3643c3cff8523/install_winelink.sh )
You can then download the older script to your pi with `wget https://raw.githubusercontent.com/WheezyE/Winelink/5d4d6f83023eb9e7b02ec783a9b3643c3cff8523/install_winelink.sh && bash install_winelink.sh`

There's a bug in versions of box86 after Dec 2021 related to VARA not using TCP/IP correctly that is preventing me from updating it in future iterations of the script. I have to find that bug and see if Seb can fix it before I can try to squash other bugs.

**Comment by davirond** [comment]
Ubuntu 24.04 - Icom IC-705 => ok
Hi, my setup is very close to have everything working. Prerequisite : PC/Rig integration works with linux-native software (WSJT-X ...), that clears all access right issues, identify which tty device, audio device etc ...
PC : Lenvovo Thinkpad L14 gen2 i5 - 16Gb RAM
Ubuntu 24.04 (current kernel 6.11.0-21-generic)
Installation through Winelink script ran fine.
Every modem/TNC is configured with CAT. My ic705 mimics an IC7300 and is declared as such in all CAT configurations. I changed the CI-V value from A4h to 94h in the rig as I once had a software which didn't know the IC-705. That causes no issue so far.

Here are the issues I found at that point :

1. minor issue : Vara FM knows only up to COM16, whereas  tty devices for my IC-705 are mapped respectively on COM33 and COM34. I found two workarounds :
- recreate the symoblic link (ln -s /dev/ttyACM0 ~/.wine/dosdevices/COM4) but it is frequently overriden for some reason, so it becomes annoying
- or edit VARAFM.ini file in the VARA FM directory and set COM33 as COM port for both CAT and PTT.

2. minor issue : the winlinkexpress.desktop file in /usr/share/applications is here to have an icon for the application in the applications menu. However it won't work. Error message is "Exec field has not been defined" which is not the case. Current workaround is that I launch the program from the CLI.

3. One more important issue for me is that templates/form do not work. When I try to call any form, I get an error message : "Unable to open template file". I also get the below message in the console. 
`03b8:fixme:uiautomation:UiaReturnRawElementProvider UIA-to-MSAA bridge not implemented, no provider map to free.`
Templates are important for me as my EmComm org is using this feature. Anyone has an idea how to get this solved ?

Thanks

**Reply by SpudGunMan** [reply]
Use regedit (via winetricks) to set the serial don't use the old symlink it's deprecated. 

Forums work here, can't replicate error. (Latest wine on mint-Intel)

**Reply by davirond** [reply]
thanks for the tip. So far I am happy with my workaround but will surely give a try.

Concerning the forms not working, I found the root cause : everything is install in a directory which name starts with a "." (~/.wine). In Ubuntu distribution, all browsers are provided as snap, which prevent applications running that way from accessing resources in home directory that starts with a ".". I deinstalled firefox from the ubuntu repository and installed it from the mozzilla repository. And voila !
And by the way, the error message that I mentioned above seems not linked at all with this issue as I figured out that it appears many time without playing with forms. So this is about something else. But so far it didn't affect any feature as far as I could observe.

Still investigating the .desktop file. I found that spaces in path should be represented in escape character (\s in this case) but this doesn't seem to be of any help.

Thanks
